### ISSUE TITLE
1km entry forms for support use & expert use

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/PoMS#118]
Martin to send links to equivalent pages on iRecord

**Comment by andrewvanbreda** [comment]
These needs to be separate forms as they include extra species grids not available on standard forms

**Comment by kitenetter** [comment]
The pages linked below are unpublished, and show no data (because the data has been moved to the new website).

Page that allowed existing samples to be selected and edited (available to all with 1 km square role):
https://www.brc.ac.uk/irecord/node/7119?group_id=1073&implicit=f
https://www.brc.ac.uk/irecord/node/7119/edit

Page that allowed existing samples to be selected, and insect 'bycatch' counts added (available to UKCEH staff role):
https://www.brc.ac.uk/irecord/node/7157?group_id=1073&implicit=f
https://www.brc.ac.uk/irecord/node/7157/edit

Page that allowed existing samples to be selected, and insect species data added (availabel to UKCEH staff and Consultant entomologists roles):
https://www.brc.ac.uk/irecord/node/7165?group_id=1073&implicit=f
https://www.brc.ac.uk/irecord/node/7165/edit

**Comment by andrewvanbreda** [comment]
Hi @kitenetter 

These two reports link to two pages but I can't see them. It says access denied but I am an iRecord admin, have these been deleted? If not I will need exact details of the extra grids that were on the page so I can add them to UKPoMS

https://www.brc.ac.uk/irecord/node/7158
https://www.brc.ac.uk/irecord/node/7163

**Comment by kitenetter** [comment]
@andrewvanbreda I can access them in edit view, does that work for you?:
https://www.brc.ac.uk/irecord/node/7158/edit
https://www.brc.ac.uk/irecord/node/7163/edit

**Comment by andrewvanbreda** [comment]
@kitenetter 
Yes that is working thanks, should be enough

**Comment by andrewvanbreda** [comment]
Hi @kitenetter @CCarvell ,

I have setup these two pages (and associated Edit pages)

https://ukpoms.org.uk/select-samples-bycatch-data
https://ukpoms.org.uk/select-samples-species-data

I have not made them available in menu yet, as am not sure if you are happy with them.

@CCarvell, I think Martin is on leave, so I don't know if you have an opinion on the points I make in this comment.

Some changes I made from the iRecord versions

- Fixed the Date field which wasn't read only
- Have removed the Delete button so the sample cannot be accidentally removed by an expert.
- The columns on the two report pages are now more consistent with each other (actually am not sure the Pan Number column would of filled-in on one of the reports the way they were previously setup, it wasn't on the new website anyway until I fixed it). 

Some questions
1. Where in the menu system you want the links to go.

2. Currently the report pages list they are for data for 2020, should this be changed to say 2021?
Also the report is setup to show data from 01/01/2019, should this be changed?

3. I notice the comment field on both data entry pages currently tell the support/expert user they can add to the comment, but they should not delete the data.
I am wondering actually if the comment should be read-only and the support comment should go in a separate attribute.
This would also make sense as we can see who is making the different comments rather than them being jumbled up.
Although if this is reported anywhere other than the data entry form itself, this needs taking into account.

Andy

**Comment by CCarvell** [comment]
@andrewvanbreda thank you very much for this update and useful summary of the questions remaining.

I agree with what you are suggesting above but think it would be best to wait for Martin's @kitenetter return to check the answers.
If this is for entry of the by-catch data for 2021 (which will mostly be done by Nadine in the lab later in the season) then for the species-specific data entry (which will be done by the taxonomists after Nadine's lab sorting) then the page is not urgently needed by any volunteers or team members.

**Comment by kitenetter** [comment]
@andrewvanbreda answers to your questions:

1. These forms should be available under "Add your results". 
The select-samples-bycatch-data one should be accessible only to these roles: admin; UKCEH staff
The select-samples-species-data one should be accessible only to these roles: admin; UKCEH staff; consultant entomologists

2. We don't need to have the year in the page titles - I have edited the text to remove the references to 2020.
Is it possible to add a dropdown box to allow filtering by year, similar to the one that appears on the map of FIT Counts (it should default to the current year):

![image](https://user-images.githubusercontent.com/16039111/131164790-2f3df87d-ed21-4654-926c-0db68b85ead4.png)

3. I agree that there is an argument for making the comment box more user-specific on the bycatch page, but we rarely need to add additional comments, and where we do we can advise staff to add their initials for clarity, so I think it is fine to leave it as it is. The only comments on the species page are linked to the occurrence rows, and it is fine to leave them as they are.

**Comment by kitenetter** [comment]
Additional question - on the enter species form (i.e. the form you get to from the edit button on select-samples-species-data), we filter the UK Master List so that it only allows entry of Hymenoptera and Diptera:

![image](https://user-images.githubusercontent.com/16039111/131165389-60b9d004-a6f0-411d-966a-de44116c5115.png)

I would prefer to limit this further so that it only allows selection of species that are bees or hoverflies. Is there any way of doing that? Bees are in a number of different familes within the Hymenoptera, hoverflies are a single family within the Diptera.

**Comment by andrewvanbreda** [comment]
@kitenetter 
1. OK
2. OK
3. OK

About the filter I don't know I will have a quick look. We can't mix types of filter, so they would have to be limited by Taxon Group, but I need to look in the database. The taxon group situation is not actually that simple as I see some are duplicated and spelt differently so that makes things even more confusing for me as each taxon only has one taxon group.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Whenever you get a sec Biren, could you pull these reports in the PoMS reports folder on live. Cheers

select_samples_to_add_insect_data_mapping
select_samples_to_add_insect_data
squares_summary_mappable

**Comment by BirenRathod** [comment]
@andrewvanbreda I have now pulled those reports now.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Thanks Biren :)

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Sorry Biren, I just realised I made a mistake, one of the reports I asked for wasn't the right one, could you pull this please thanks.
Thanks
get_years_for_population_call

**Comment by BirenRathod** [comment]
@andrewvanbreda 
Have you committed in the master branch? I don't see any reports pulling down.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
I saw it seems to have pulled anyway, it seem to be there much to my surprise.
I  spotted a problem, so will need a re-pull later anyway, will let you know.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Hi Biren,

I fixed those problems, if you could pull the following whenever you get a chance. Cheers
select_samples_to_add_insect_data_mapping
select_samples_to_add_insect_data
get_years_for_population_call

**Comment by BirenRathod** [comment]
@andrewvanbreda Reports have pulled down now.

**Comment by andrewvanbreda** [comment]
AVB following notes are for myself.
I have tested this successfully (inc save, permissions). All that is needed to go live is to enable the menu items.
Just two things to address
1. Martin's filter question
2. There is a problem when editing records on these pages as system does not allow editing of records you didn't create currently.

**Comment by andrewvanbreda** [comment]
AVB: Note to myself
Problem 2 fixed.

**Comment by andrewvanbreda** [comment]
Hi @kitenetter 

In terms of your question about filtering.
Without spending extra budget on coding, the only option would be to supply me with an extact list of species to limit to (this could be any of the following->preferred name, taxon meaning ID, taxa_taxon_list_id,taxon_id,external_key)

Unfortunately, this filtering option does not seem to currently allow you to specificy a taxon in the taxon hierarchy (e.g. a Family) and then pick up all the children of that item.
I would need to code a new option for Indicia (which obviously outside the scope of the current budget).

So basically the option you have to to provide me with an exact list of species to limit to.

Hope that helps

Andy

**Comment by andrewvanbreda** [comment]
@kitenetter 
This is now live.
Please close if happy.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Another quick not about this. The Year filter on the page will go back to all project year (2017), actually there is a background limiter that has come across from iRecord that actually limits the page to 2019 and later.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Sorry, another quick note for you to check.
I noticed the two data entry forms for this have this configuration
occurrence:record_status=C
occurrence:release_status=U

Is it correct these forms should be setting unreleased status, this has come across from the old iRecord form configuration.

**Comment by kitenetter** [comment]
@andrewvanbreda we needed these settings in iRecord to ensure that the PoMS data didn't get exposed in full detail to the public and to the verifiers. Now that the PoMS website is separate, and doesn't have an agreement with iRecord, we no longer need to flag the occurrences as confidential nor unreleased.

Question: in the warehouse, the Pollinator Monitoring Scheme website is linked to the test URL. Is it okay for me to update that URL? We might as well have it correct for the live website (but I'm not sure if there are any other consequences from changing the URL stored in the warehouse).

**Comment by BirenRathod** [comment]
@kitenetter It is fine to update on both warehouse. it will not impact as long as id is same.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Have removed the Unreleased configuration lines. Do you know if anyone would of used these pages on the site yet?

**Comment by kitenetter** [comment]
I don't think so - I haven't had a chance to check them yet, and haven't told anyone else about them, so I would be surprised if any data has been entered via thir route as yet.

**Comment by kitenetter** [comment]
@andrewvanbreda I'm looking at the form for adding 'bycatch' data:
https://ukpoms.org.uk/node/43/edit

This is currently using species list ID 221 "Insect groups (PoMS)". It needs to change to species list ID 234 "Insect groups (PoMS pan-traps)".

At the moment species list ID 234 is linked to iRecord and is not available to use on the PoMS website. Can we just change the "Owned by" value from iRecord to "<Warehouse>"? And then select this list for the bycatch form?

**Comment by andrewvanbreda** [comment]
@kitenetter 
Done.
Please check.

**Comment by kitenetter** [comment]
All seems well with these forms, thanks. Closing.
